server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -456,6 +448,31 @@ public DeployDestination planDeployment(VirtualMachineProfile vmProfile, Deploym
         return dest;
     }
 
+    private Host checkDeployInVmLastHost(VirtualMachineProfile vmProfile, VirtualMachine vm) {
+        String considerLastHostStr = (String)vmProfile.getParameter(VirtualMachineProfile.Param.ConsiderLastHost);
+        String haVmTag = (String)vmProfile.getParameter(VirtualMachineProfile.Param.HaTag);
+        boolean considerLastHost = vm.getLastHostId() != null && haVmTag == null &&
+                (considerLastHostStr == null || Boolean.TRUE.toString().equalsIgnoreCase(considerLastHostStr));
```

Comment:
Hi @sureshanaparti I'm not clear why the first part of this OR operator would make it consider the last host. I would assume if the ConsiderLastHost parameter is not set that this evaluates to false?

Reply:
@nvazquez by default last host is considered if not explicitly set. same check is there earlier, and no changes to it expect moving the condition to a separate method.